### ISSUE TITLE
(re-)load a project

### ISSUE THREAD

**Issue by mooseyboots** [bbatsov/projectile#1838]
this is more just a query about how to re-load a project. i wonder if this functionality is implemented, and if not, whether users have some other way of handling this. i'm only a very basic user of projectile.el.

when working on an elisp project, i often need to change branches, then re-load my project's code. i.e. i switch to dired, nav to the code dir, call `t` to invert all marks (mark all if none marked), then call `L` to load all marked files.

does projectile.el have a way to do this?

**Comment by bbatsov** [comment]
Does this help with what you're trying to achieve?

```
(global-auto-revert-mode t)
```

Projectile currently can close Projectile buffers, but there's no functionality to recreate special buffers.

**Comment by mooseyboots** [comment]
i don't think so, no. i'm not worried about whether buffers are open or not actually. i'm interested in loading the code inside them after switching git branches. i work on a branch in an elisp project, but then i want to return to say develop branch for actually using the project in my own emacs set up. so i save/close the project's buffers, switch branches in magit, but the feature branch is still the code that is loaded into my emacs session. so i have to manually go to the code in dired and load it with `dired-do-load` as i mentioned.

it's no problem if it's not in projectile, i just wondered also how others deal with this issue. i guess it's only an issue for elisp devs.

**Comment by mooseyboots** [comment]
i cooked up a hack to do this, but it isn't general, it just loads `.el` files:

```
(defun mb/projectile-load-source-files ()
    ""
    (interactive)
    (let* ((projectile-root (projectile-acquire-root))
           (subdir (read-directory-name "Load dir: " projectile-root))
           (regex (rx bos
                      (not ".") ; don't load hidden files
                      (* (any alnum "-" "_"))
                      ".el"
                      eos))
           (files (directory-files subdir nil regex)))
      (dolist (file files)
        (load file))))
```

**Comment by bbatsov** [comment]
Yeah, I think your use-case is quite specific to Lisps, so I'd say it's outside the scope of Projectile's core functionality.

CIDER (for Clojure) implements similar reloading functionality to what you describe, although in a more sophisticated manner as just reloading the code is not enough to clear the old definitions and those can cause some issues.
